src/makeCalculator.js

```diff
@@ -4,7 +4,49 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculate = {
+    result: 0,
+
+    add(number) {
+      this.result += number;
+    },
+
+    subtract(number) {
+      this.result -= number;
+    },
+
+    multiply(number) {
+      this.result *= number;
+    },
+
+    divide(number) {
+      if (number === 0) {
+        console.log('Cannot divide by zero');
+      }
+
+      this.result /= number;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(callback, number) {
+      const isCalculatorMethod = [this.add, this.subtract, this.multiply, this.divide].includes(callback);
```

Comment:
Checking if the callback is one of the calculator methods is unnecessary. Since `operate` is a public method, it should trust that the provided callback is a function and simply invoke it. Remove the array check to simplify the code.